src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageUtility.java

```diff
@@ -0,0 +1,110 @@
+package ca.corefacility.bioinformatics.irida.repositories.filesystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+
+import ca.corefacility.bioinformatics.irida.exceptions.ConcatenateException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Interface describing methods for performing storage actions
+ */
+
+public interface IridaFileStorageUtility {
+	//Valid extensions to try to concatenate with this tool
+	public static final List<String> VALID_EXTENSIONS = Lists.newArrayList("fastq", "fastq.gz");
+	/**
+	 * Get a temporarry file from storage
+	 *
+	 * @param file The {@link Path} to the file
+	 * @return {@link File} which was retrieved from path
+	 */
+	public File getTemporaryFile(Path file);
+
+	/**
+	 * Get file size in bytes
+	 *
+	 * @param file The {@link Path} to the file
+	 * @return {@link Long} size of file retrieved from path
+	 */
+	public Long getFileSize(Path file);
+
+	/**
+	 * Write file to storage (azure, aws, or local)
+	 *
+	 * @param source The {@link Path} to the file
+	 * @param target The {@link Path} to where file should be moved
+	 * @param sequenceFileDir The {@link Path} to sequence file directory
+	 * @param sequenceFileDirWithRevision The {@link Path} to sequence file revision directory
+	 */
+	public void writeFile(Path source, Path target, Path sequenceFileDir, Path sequenceFileDirWithRevision);
+
+	/**
+	 * Returns if the storage type is local or not
+	 *
+	 * @return {@link Boolean#TRUE} if local, {@link Boolean#FALSE} if not.
+	 */
+	public boolean storageTypeIsLocal();
```

Comment:
This isn't used.  Is it used in later branches?

Reply:
I think it was used in some later branches. If not I will remove in a future branch